### ISSUE TITLE
[Enhancement] Disable UseSecondary by default on RawDataTooltip

### ISSUE THREAD

**Issue by Hayden-da-Serg** [Frooxius/NeosPublic#607]
I have found many cases in which UseSecondary is enabled on a RawDataTooltip when this is very much not necessary, therefore locking that thumbstick locomotion on non-Index controllers, I myself am even guilty of this on occasion.

Having this enabled by default may also be confusing to an unexperienced user that doesn't know why their tool is locking their locomotion.

A good solution to this would be to have UseSecondary disabled by default on RawDataTooltip, and if the user specifically needs that tool to lock locomotion for whatever reason, they can then go enable it.

**Comment by GearBell** [comment]
i agree itd be very helpful with workflow.

**Comment by Frooxius** [comment]
This is tricky, because if it's off by default, but you use it, the tool will still be confusing to use, as using its secondary will also make the player move.

This is something that the creator should think about setting in either case. Having it default on has the benefit that it causes immediate complaints or issues as the user can't move, making the creator think about the option.

**Comment by shiftyscales** [comment]
I imagine though, that if it's off by default, the same users could instead note that they're moving/jumping while trying to use secondary. So the creator will get the same feedback either way.

Personally, I'm all in favor of any change that will limit how often a user loses access to their ability to turn/move with their analog stick on Oculus Touch/other controllers.

I still believe it could be beneficial to split the "secondary" into separate click/axis functions. "UseSecondary" and "UseAxis".

This would allow the use of a secondary button without disabling locomotion via the axis unless 'UseAxis' was also enabled, enabling the flexibility to have more tools, toys, and gadgets that allow near-full locomotion on Oculus Touch controllers without any greater added confusion, or inconsistency than already exists when users lose access to their locomotion (yet don't always because of the nature of the UseSecondary option).

**Comment by Frooxius** [comment]
I don't think that's always the case. It might seem as a more normal behavior to some, whereas completely losing control is more harsh.

In my experience having this approach where it's less "fuzzy" is more beneficial overall, because it has much more likelyhood of pushing you to change.

It is not possible to split "UseSecondary" and "UseAxis", because on certain controllers and control schemes this is coupled and would lead to severe complications and confusion to user.

**Comment by shiftyscales** [comment]
While physically, they aren't separable, they are separably addressable, E.g. in the case of SteamVR input, a touchpad click is distinct from a touchpad touch, and you can remap them. E.g. with my Index controller, I remapped the joystick click to the "A" button.

If a user swapped their inputs, it would no longer make sense, e.g. say a Vive user swapped their menu button with the touchpad click. Their menu button would be blocked even though it's a completely separate input.

If we ever decide we want to handle native rebinding, it makes a lot more sense to address every input separably rather than linking them together based on their physical proximity.

**Comment by Frooxius** [comment]
That doesn't help us at all. We're dealing with them in Neos, not outside.

**Comment by shiftyscales** [comment]
Exactly- hence if we ever decide we _do_ want to deal with them in Neos, it severely restricts some potential. If the tools are set-up correctly it allows additional granularity without being any more confusing than it already is- while giving more control both figuratively, and literally.

There are already cases where users are confused by the inability to move while they have a tool equipped- especially since now it's possible they _can_ move with a tool equipped depending on the state of "UseSecondary". Existing instances of tools could be upgraded to have both "UseSecondary" and "UseAxis" enabled and it would function exactly as it does now, while creators can define how they wish to use those inputs on new tools made.

Going forward- having access to locomotion while still having the secondary click used for a tool's function also addresses the core issue this issue was created in response to.

The reason it's desired to have "UseSecondary" disabled by default is so that by default the tool will provide locomotion on the stick for Oculus Touch/Vive users by default.

**Comment by Frooxius** [comment]
You're missing the point. It doesn't matter that they are decoupled inputs coming from the controller, we cannot decouple them like that in the input system without making a huge mess.

Locomotion modules and tools can use secondary each variety of different ways and methods. Furthermore some controllers have those two inherently coupled. 

All of that combined leads to potentially unlimited number of potential combinations that both the code and user would have to deal with, because now each tool needs to consider all possible ways locomotion module can utilize the input (across different controllers at that) and each locomotion module also needs to know how the tools are utilizing it, so they can mesh well together. 

With a simple binary choice, it's simple. Either locomotion control has input or the tool does. Doesn't matter which controller it is. Does matter which control scheme either uses.

It's just not worth the significant complications from development and user interaction perspective. 

We've already discussed this, I don't see point in rehashing it again.

**Comment by shiftyscales** [comment]
You're right, I'm sorry, @Frooxius.
